enterprise/dbcrypt/cliutil.go

```diff
@@ -212,3 +214,25 @@ func Delete(ctx context.Context, log slog.Logger, sqlDB *sql.DB) error {
 
 	return nil
 }
+
+// allUserIDs returns _all_ user IDs we know about, regardless of status or deletion.
+// We need to encrypt / decrypt tokens regardless of user status or deletion as they
+// may still be valid. While we could check the expiry, we also don't know if the
+// provider is lying about expiry.
+// This function will likely only ever be used here, so keeping it here instead
+// of exposing it in all of our database-related interfaces.
+func allUserIDs(ctx context.Context, sqlDB *sql.DB) ([]uuid.UUID, error) {
+	var id uuid.UUID
+	userIDs := make([]uuid.UUID, 0)
+	rows, err := sqlDB.QueryContext(ctx, `SELECT DISTINCT id FROM users`)
```

Comment:
Second case for `testqueries` 🤩

Reply:
This isn't a test query though, this is the real deal.

Reply:
Ooh, sorry, I simply assumed since it was inlined in code. Why not create a query in `queries/*.sql` for this?

Reply:
And now that I read the comment more clearly, I see why. I think it'd be fine to expose though. I think there are many queries we only use in one place.

Reply:
Is there some way we can be extremely sure that this function only gets called from CLI with a direct DB connection?  It bypasses authz, so it's important that coderd isn't able to call it.

Reply:
> If that's a worry, I can just add it as a regular query with all of the associated authz stuff. 

Right now it's not exported from the `dbcrypt` package. If it being exported down the line is a worry, I can just add it as a regular query with all of the associated authz stuff instead.